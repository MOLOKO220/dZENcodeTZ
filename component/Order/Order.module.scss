.Order {
  min-height: 60px;
  width: 90vw;
  border: solid 1px rgb(191, 191, 191);
  border-radius: 10px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 5px;
  transition: 300ms;
  @media (max-width: 720px) {
    flex-direction: column;
    padding: 10px;
  }
  @media (max-width: 500px) {
    width: 85vw;
  }
  &:hover {
    box-shadow: 5px 5px 10px rgba(0, 0, 0, 0.2);
  }
  > div {
    display: flex;
    justify-content: space-around;
    align-items: center;
  }
  > div:nth-child(1) {
    width: 40%;
    @media (max-width: 720px) {
      width: 100%;
      border-bottom: 1px solid gray;
      margin-bottom: 5px;
      padding-bottom: 5px;
    }
    h4 {
      width: auto;
      font-family: "Roboto", sans-serif;
      color: gray;
      overflow: hidden;
    }
  }
  > div:nth-child(2) {
    width: 60%;
    @media (max-width: 720px) {
      width: 100%;
    }
    .Order__burgeBtn {
      height: 40px;
      width: 40px;
      cursor: pointer;
      transition: 200ms;
      &:hover {
        transform: scale(1.1);
      }
      &:active {
        transform: scale(0.9);
      }
    }
    .Order__product {
      font-family: "Roboto", sans-serif;
      display: flex;
      flex-direction: column;
      align-items: center;
      > div {
        width: 30px;
        text-align: center;
        font-size: 16px;
        color: black;
      }
      > p {
        font-size: 12px;
        color: gray;
      }
    }
    .Order__date {
      font-family: "Roboto", sans-serif;
      display: flex;
      flex-direction: column;
      align-items: center;
      @media (max-width: 450px) {
        display: none;
      }
      > div {
        width: 40px;
        font-size: 12px;
        color: gray;
      }
      > p {
        font-size: 14px;
        color: black;
      }
    }
    .Order__prices {
      font-family: "Roboto", sans-serif;
      > div {
        width: 40px;
        font-size: 12px;
        color: gray;
      }
      > p {
        font-size: 14px;
        color: black;
      }
    }
    .Order__delete_btn {
      height: 30px;
      width: 30px;
      cursor: pointer;
      transition: 200ms;
      &:hover {
        transform: scale(1.3);
      }
      &:active {
        transform: scale(0.9);
      }
    }
  }
  .Order__removePopup {
    width: 100vw;
    height: 100vh;
    background: rgba(0, 0, 0, 0.4);
    position: fixed;
    top: 0;
    right: 0px;
    z-index: 800;
    display: none;
    align-items: center;
    justify-content: center;
    > div {
      height: 200px;
      width: 400px;
      background: white;
      border-radius: 10px;
      overflow: hidden;
      position: relative;
      @media (max-width: 515px) {
        width: 300px;
      }
      @media (max-width: 380px) {
        width: 260px;
        margin-top: 10px;
        margin-left: 50px;
      }
      header {
        height: 30%;
        border-bottom: solid 1px rgba(0, 0, 0, 0.2);
        display: flex;
        align-items: center;
        padding: 15px;
        > h5 {
          font-family: "Roboto", sans-serif;
          font-size: 16px;
        }
      }
      main {
        height: 30%;
        border-bottom: solid 1px rgba(0, 0, 0, 0.2);
        display: flex;
        align-items: center;
        justify-content: center;
        > h6 {
          font-family: "Roboto", sans-serif;
          font-size: 18px;
          font-weight: 500;
        }
      }
      footer {
        height: 40%;
        background: rgb(124, 179, 66);
        display: flex;
        align-items: center;
        justify-content: flex-end;
        padding: 20px;
        > button {
          font-family: "Roboto", sans-serif;
          font-size: 18px;
          font-weight: 400;
          transition: 300ms;
          &:hover {
            transform: scale(1.1);
          }
        }
        > button:nth-child(1) {
          background: none;
          color: white;
        }
        > button:nth-child(2) {
          color: white;
          background: red;
          padding: 5px;
          border-radius: 5px;
          margin-left: 15px;
        }
      }
      .Order__removePopup__closeBtn {
        background: none;
        display: flex;
        align-items: center;
        justify-content: center;
        position: absolute;
        right: 0px;
        top: 0;
        height: 40px;
        width: 40px;
        cursor: pointer;
        transition: 300ms;
        &:hover {
          transform: scale(1.1);
        }
      }
    }
  }
  .removePopup__active {
    display: flex !important;
  }
}
