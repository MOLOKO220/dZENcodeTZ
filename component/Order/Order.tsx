import React, { useEffect, useRef, useState } from "react";
import Link from "next/link";
import style from "./Order.module.scss";
import { useDispatch } from "react-redux";
import { removaOrder } from "../../storage/mainReducer";

interface propsType {
  data: {
    id: number;
    title: string;
    date: string;
    description: string;
    products: any;
  };
}
export default function Order(props: propsType) {
  // hooks
  // Popup
  const RemovePopup = useRef<HTMLDialogElement>(null);

  //Prices
  const [USD, setUSD] = useState<number>(0);
  const [UAH, setUAH] = useState<number>(0);

  useEffect(() => {
    // перебрать все продукты и сумируем цены
    let allUsd = 0;
    let allUah = 0;
    props.data.products.map((el: any) => {
      allUsd = allUsd + el.price[0].value;
      allUah = allUah + el.price[1].value;
    });
    setUSD(allUsd);
    setUAH(allUah);
  }, []);

  // redux
  const dispatch = useDispatch();

  // Popup
  function showRemovePopup() {
    RemovePopup.current?.classList.add(style.removePopup__active);
  }
  function closeRemovePopup() {
    RemovePopup.current?.classList.remove(style.removePopup__active);
  }
  // удаляем заказ
  function deleteOreder() {
    dispatch(removaOrder(props.data.id));
  }

  //Date
  function dateFormat(date: string, format: number) {
    const d = new Date(date);
    // 1  формат
    if (format === 1) {
      // принимаем дату и приводим её в цифровой формат: дд/мм/гг
      // добавляем 0 перед датой если она ниже 10, чтобы получить "04" и тд
      return `${d.getDate() < 10 ? "0" + d.getDate() : d.getDate()}/${
        d.getMonth() + 1 < 10 ? "0" + (d.getMonth() + 1) : d.getMonth() + 1
      }/${d.getFullYear()} `;
    }
    // 2 формат
    if (format === 2) {
      // принимаем дату и приводим её в формат: дд/МЕСЯЦ/гг
      return `${
        d.getDate() < 10 ? "0" + d.getDate() : d.getDate()
      }/${d.toLocaleString("ru", { month: "long" })}/${d.getFullYear()}`;
    }
    // 3 формат
    if (format === 3) {
      // принимаем дату и приводим её в цифровой формат: дд/мм
      return `${d.getDate() < 10 ? "0" + d.getDate() : d.getDate()}/${
        d.getMonth() + 1 < 10 ? "0" + (d.getMonth() + 1) : d.getMonth() + 1
      }`;
    }
  }

  return (
    <li className={style.Order}>
      <div>
        <h4>{props.data.title}</h4>
      </div>
      <div>
        <Link href={`/${props.data.id}`}>
          <img
            className={style.Order__burgeBtn}
            src="burgeBtn.png"
            alt="show more"
          />
        </Link>
        <div className={style.Order__product}>
          <div>{props.data.products.length}</div>
          <p>Продукта</p>
        </div>
        <div className={style.Order__date}>
          <div>{dateFormat(props.data.date, 3)}</div>
          <p>{dateFormat(props.data.date, 2)}</p>
        </div>
        <div className={style.Order__prices}>
          <div>{USD} $</div>
          <p>{UAH} UAH</p>
        </div>
        <img
          className={style.Order__delete_btn}
          src="trashCan.png"
          alt="Delete"
          onClick={showRemovePopup}
        />
      </div>
      <dialog className={style.Order__removePopup} ref={RemovePopup}>
        <div>
          <header>
            <h5>Вы уверены, что хотите удалить этот приход?</h5>
          </header>
          <main>
            <h6>{props.data.title}</h6>
          </main>
          <footer>
            <button onClick={closeRemovePopup}>ОТМЕНИТЬ</button>
            <button onClick={deleteOreder}>УДАЛИТЬ</button>
          </footer>
          <img
            className={style.Order__removePopup__closeBtn}
            src="closeBtn.webp"
            alt="closeBtn"
            onClick={closeRemovePopup}
          />
        </div>
      </dialog>
    </li>
  );
}
