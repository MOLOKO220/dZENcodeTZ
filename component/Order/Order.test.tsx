import { render, screen } from "@testing-library/react";
import Order from "./Order";

const mockDispatch = jest.fn();
const mockSelector = jest.fn();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
  useSelector: () => mockSelector,
}));

describe("Order", () => {
  it("render", () => {
    render(
      <Order
        data={{
          id: 2,
          title: "Order1",
          date: "03.11.2017",
          description: "string",
          products: [],
        }}
      />
    );

    expect(
      screen.getByText("Вы уверены, что хотите удалить этот приход?")
    ).toBeInTheDocument();
  });

  it("date", () => {
    render(
      <Order
        data={{
          id: 2,
          title: "Order1",
          date: "03.11.2017",
          description: "string",
          products: [],
        }}
      />
    );

    expect(screen.getByText("11/март/2017")).toBeInTheDocument();
  });

  it("snapshots", () => {
    const component = render(
      <Order
        data={{
          id: 2,
          title: "Order1",
          date: "03.11.2017",
          description: "string",
          products: [],
        }}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
