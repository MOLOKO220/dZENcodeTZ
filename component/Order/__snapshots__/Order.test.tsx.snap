// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order snapshots 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="Order"
      >
        <div>
          <h4>
            Order1
          </h4>
        </div>
        <div>
          <a
            href="/2"
          >
            <img
              alt="show more"
              class="Order__burgeBtn"
              src="burgeBtn.png"
            />
          </a>
          <div
            class="Order__product"
          >
            <div>
              0
            </div>
            <p>
              Продукта
            </p>
          </div>
          <div
            class="Order__date"
          >
            <div>
              11/03
            </div>
            <p>
              11/март/2017
            </p>
          </div>
          <div
            class="Order__prices"
          >
            <div>
              0
               $
            </div>
            <p>
              0
               UAH
            </p>
          </div>
          <img
            alt="Delete"
            class="Order__delete_btn"
            src="trashCan.png"
          />
        </div>
        <dialog
          class="Order__removePopup"
        >
          <div>
            <header>
              <h5>
                Вы уверены, что хотите удалить этот приход?
              </h5>
            </header>
            <main>
              <h6>
                Order1
              </h6>
            </main>
            <footer>
              <button>
                ОТМЕНИТЬ
              </button>
              <button>
                УДАЛИТЬ
              </button>
            </footer>
            <img
              alt="closeBtn"
              class="Order__removePopup__closeBtn"
              src="closeBtn.webp"
            />
          </div>
        </dialog>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="Order"
    >
      <div>
        <h4>
          Order1
        </h4>
      </div>
      <div>
        <a
          href="/2"
        >
          <img
            alt="show more"
            class="Order__burgeBtn"
            src="burgeBtn.png"
          />
        </a>
        <div
          class="Order__product"
        >
          <div>
            0
          </div>
          <p>
            Продукта
          </p>
        </div>
        <div
          class="Order__date"
        >
          <div>
            11/03
          </div>
          <p>
            11/март/2017
          </p>
        </div>
        <div
          class="Order__prices"
        >
          <div>
            0
             $
          </div>
          <p>
            0
             UAH
          </p>
        </div>
        <img
          alt="Delete"
          class="Order__delete_btn"
          src="trashCan.png"
        />
      </div>
      <dialog
        class="Order__removePopup"
      >
        <div>
          <header>
            <h5>
              Вы уверены, что хотите удалить этот приход?
            </h5>
          </header>
          <main>
            <h6>
              Order1
            </h6>
          </main>
          <footer>
            <button>
              ОТМЕНИТЬ
            </button>
            <button>
              УДАЛИТЬ
            </button>
          </footer>
          <img
            alt="closeBtn"
            class="Order__removePopup__closeBtn"
            src="closeBtn.webp"
          />
        </div>
      </dialog>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
